python/src/clexulator.cpp

```diff
@@ -1161,7 +1161,22 @@ PYBIND11_MODULE(_clexulator, m) {
               Change in extensive correlations, relative to `reference_extensive_correlations`.
           )pbdoc",
            py::arg("key"), py::arg("new_value"),
-           py::arg("`reference_extensive_correlations"));
+           py::arg("`reference_extensive_correlations"))
+      .def("grad_correlations", &clexulator::Correlations::grad_correlations,
+           py::return_value_policy::reference_internal, R"pbdoc(
+          Calculate and return gradients of correlations
+
+          Parameters
+          ----------
+          key: str
+              Specifies the type of DoF
+
+          Returns
+          -------
+          gradients_of_correlations: np.ndarray
```

Comment:
Can you please improve this docstring? 

For the `key` parameter, let's use `dof_key` and copy the text from one of the other methods.

Definitely need to describe the `gradients_of_correlations` array more: at least dimensions, dtype, and row/column meaning. For the type, I prefer: `numpy.ndarray[np.float64[m,n]]`, where `m` and `n` (or a more descriptive variable name) are described in the text.